src/main/java/net/dv8tion/jda/internal/entities/VoiceChannelImpl.java

```diff
@@ -136,10 +148,37 @@ public VoiceChannelImpl setUserLimit(int userLimit)
         return this;
     }
 
+    public VoiceChannelImpl setNSFW(boolean nsfw)
+    {
+        this.nsfw = nsfw;
+        return this;
+    }
+
     // -- Abstract Hooks --
+
     @Override
     protected void onPositionChange()
     {
         getGuild().getVoiceChannelsView().clearCachedLists();
     }
+
+    @Override
+    public boolean canTalk(@Nonnull Member member)
+    {
+        Checks.notNull(member, "Member");
+        return member.hasPermission(this, Permission.MESSAGE_SEND);
+    }
+
+    @Override
+    public long getLatestMessageIdLong()
+    {
+        return latestMessageId;
+    }
+
+    @Override
+    public VoiceChannelImpl setLatestMessageIdLong(long latestMessageId)
+    {
+        this.latestMessageId = latestMessageId;
+        return this;
+    }
```

Comment:
Move these somewhere above the  // -- Abstract Hooks -- to be closer to the impl getters or impl setters as needed for consistent grouping